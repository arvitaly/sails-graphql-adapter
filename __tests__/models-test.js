"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator.throw(value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments)).next());
    });
};
const sails_fixture_app_1 = require("sails-fixture-app");
const models_1 = require("./../models");
describe("Model tests", () => {
    let app;
    beforeEach(() => __awaiter(this, void 0, void 0, function* () {
        app = yield sails_fixture_app_1.start(__dirname + "/../__fixtures__/app1");
    }));
    afterEach((done) => {
        app.lower(done);
    });
    it("model", () => {
        const models = models_1.default(app);
        expect(models).toMatchSnapshot();
    });
});
