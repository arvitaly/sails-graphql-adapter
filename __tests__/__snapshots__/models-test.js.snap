exports[`Model tests model 1`] = `
Collection {
  "modelConfigs": Array [
    Object {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "primaryKey": false,
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "key",
          "primaryKey": true,
          "realName": "key",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "primaryKey": false,
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": "model4",
          "name": "model4s",
          "primaryKey": false,
          "realName": "model4s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "primaryKey": false,
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "primaryKey": false,
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
      ],
      "id": "model2",
      "name": "Model2",
    },
    Object {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "title",
          "primaryKey": false,
          "realName": "title",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model4",
          "name": "model4s",
          "primaryKey": false,
          "realName": "model4s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "primaryKey": false,
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "primaryKey": false,
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
      ],
      "id": "model3",
      "name": "Model3",
    },
    Object {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "primaryKey": false,
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model2",
          "name": "owner",
          "primaryKey": false,
          "realName": "owner",
          "required": false,
          "type": "model",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "primaryKey": false,
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "primaryKey": false,
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "primaryKey": false,
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
      ],
      "id": "model4",
      "name": "Model4",
    },
    Object {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "primaryKey": false,
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "title",
          "primaryKey": false,
          "realName": "title",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "num",
          "primaryKey": false,
          "realName": "num",
          "required": true,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "sum",
          "primaryKey": false,
          "realName": "sum",
          "required": false,
          "type": "float",
        },
        Object {
          "model": undefined,
          "name": "isActive",
          "primaryKey": false,
          "realName": "isActive",
          "required": true,
          "type": "boolean",
        },
        Object {
          "model": undefined,
          "name": "firstActive",
          "primaryKey": false,
          "realName": "firstActive",
          "required": true,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "lastActive",
          "primaryKey": false,
          "realName": "lastActive",
          "required": false,
          "type": "date",
        },
        Object {
          "model": "model2",
          "name": "model2Field",
          "primaryKey": false,
          "realName": "model2Field",
          "required": false,
          "type": "model",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "primaryKey": false,
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "primaryKey": false,
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "primaryKey": false,
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
      ],
      "id": "modelname1",
      "name": "ModelName1",
    },
  ],
  "models": Array [
    Model {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "key",
          "primaryKey": true,
          "realName": "key",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": "model4",
          "name": "model4s",
          "realName": "model4s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
        Object {
          "name": "id",
          "realName": null,
          "required": false,
          "type": "id",
        },
      ],
      "collector": [Circular],
      "config": Object {
        "attributes": Array [
          Object {
            "model": undefined,
            "name": "name",
            "primaryKey": false,
            "realName": "name",
            "required": false,
            "type": "string",
          },
          Object {
            "model": undefined,
            "name": "key",
            "primaryKey": true,
            "realName": "key",
            "required": false,
            "type": "string",
          },
          Object {
            "model": "model3",
            "name": "model3s",
            "primaryKey": false,
            "realName": "model3s",
            "required": false,
            "type": "collection",
          },
          Object {
            "model": "model4",
            "name": "model4s",
            "primaryKey": false,
            "realName": "model4s",
            "required": false,
            "type": "collection",
          },
          Object {
            "model": undefined,
            "name": "createdAt",
            "primaryKey": false,
            "realName": "createdAt",
            "required": false,
            "type": "date",
          },
          Object {
            "model": undefined,
            "name": "updatedAt",
            "primaryKey": false,
            "realName": "updatedAt",
            "required": false,
            "type": "date",
          },
        ],
        "id": "model2",
        "name": "Model2",
      },
      "id": "model2",
      "name": "Model2",
      "opts": Object {
        "interfaces": Array [],
      },
      "primaryKeyAttribute": Object {
        "model": undefined,
        "name": "key",
        "primaryKey": true,
        "realName": "key",
        "required": false,
        "type": "string",
      },
      "queryName": "model2",
      "resolveFn": undefined,
    },
    Model {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "title",
          "realName": "title",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model4",
          "name": "model4s",
          "realName": "model4s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "_id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
        Object {
          "name": "id",
          "realName": null,
          "required": false,
          "type": "id",
        },
      ],
      "collector": [Circular],
      "config": Object {
        "attributes": Array [
          Object {
            "model": undefined,
            "name": "title",
            "primaryKey": false,
            "realName": "title",
            "required": false,
            "type": "string",
          },
          Object {
            "model": "model4",
            "name": "model4s",
            "primaryKey": false,
            "realName": "model4s",
            "required": false,
            "type": "collection",
          },
          Object {
            "model": undefined,
            "name": "id",
            "primaryKey": true,
            "realName": "id",
            "required": false,
            "type": "integer",
          },
          Object {
            "model": undefined,
            "name": "createdAt",
            "primaryKey": false,
            "realName": "createdAt",
            "required": false,
            "type": "date",
          },
          Object {
            "model": undefined,
            "name": "updatedAt",
            "primaryKey": false,
            "realName": "updatedAt",
            "required": false,
            "type": "date",
          },
        ],
        "id": "model3",
        "name": "Model3",
      },
      "id": "model3",
      "name": "Model3",
      "opts": Object {
        "interfaces": Array [],
      },
      "primaryKeyAttribute": Object {
        "model": undefined,
        "name": "_id",
        "primaryKey": true,
        "realName": "id",
        "required": false,
        "type": "integer",
      },
      "queryName": "model3",
      "resolveFn": undefined,
    },
    Model {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": "model2",
          "name": "owner",
          "realName": "owner",
          "required": false,
          "type": "model",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "_id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
        Object {
          "name": "id",
          "realName": null,
          "required": false,
          "type": "id",
        },
      ],
      "collector": [Circular],
      "config": Object {
        "attributes": Array [
          Object {
            "model": undefined,
            "name": "name",
            "primaryKey": false,
            "realName": "name",
            "required": false,
            "type": "string",
          },
          Object {
            "model": "model2",
            "name": "owner",
            "primaryKey": false,
            "realName": "owner",
            "required": false,
            "type": "model",
          },
          Object {
            "model": "model3",
            "name": "model3s",
            "primaryKey": false,
            "realName": "model3s",
            "required": false,
            "type": "collection",
          },
          Object {
            "model": undefined,
            "name": "id",
            "primaryKey": true,
            "realName": "id",
            "required": false,
            "type": "integer",
          },
          Object {
            "model": undefined,
            "name": "createdAt",
            "primaryKey": false,
            "realName": "createdAt",
            "required": false,
            "type": "date",
          },
          Object {
            "model": undefined,
            "name": "updatedAt",
            "primaryKey": false,
            "realName": "updatedAt",
            "required": false,
            "type": "date",
          },
        ],
        "id": "model4",
        "name": "Model4",
      },
      "id": "model4",
      "name": "Model4",
      "opts": Object {
        "interfaces": Array [],
      },
      "primaryKeyAttribute": Object {
        "model": undefined,
        "name": "_id",
        "primaryKey": true,
        "realName": "id",
        "required": false,
        "type": "integer",
      },
      "queryName": "model4",
      "resolveFn": undefined,
    },
    Model {
      "attributes": Array [
        Object {
          "model": undefined,
          "name": "name",
          "realName": "name",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "title",
          "realName": "title",
          "required": false,
          "type": "string",
        },
        Object {
          "model": undefined,
          "name": "num",
          "realName": "num",
          "required": true,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "sum",
          "realName": "sum",
          "required": false,
          "type": "float",
        },
        Object {
          "model": undefined,
          "name": "isActive",
          "realName": "isActive",
          "required": true,
          "type": "boolean",
        },
        Object {
          "model": undefined,
          "name": "firstActive",
          "realName": "firstActive",
          "required": true,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "lastActive",
          "realName": "lastActive",
          "required": false,
          "type": "date",
        },
        Object {
          "model": "model2",
          "name": "model2Field",
          "realName": "model2Field",
          "required": false,
          "type": "model",
        },
        Object {
          "model": "model3",
          "name": "model3s",
          "realName": "model3s",
          "required": false,
          "type": "collection",
        },
        Object {
          "model": undefined,
          "name": "_id",
          "primaryKey": true,
          "realName": "id",
          "required": false,
          "type": "integer",
        },
        Object {
          "model": undefined,
          "name": "createdAt",
          "realName": "createdAt",
          "required": false,
          "type": "date",
        },
        Object {
          "model": undefined,
          "name": "updatedAt",
          "realName": "updatedAt",
          "required": false,
          "type": "date",
        },
        Object {
          "name": "id",
          "realName": null,
          "required": false,
          "type": "id",
        },
      ],
      "collector": [Circular],
      "config": Object {
        "attributes": Array [
          Object {
            "model": undefined,
            "name": "name",
            "primaryKey": false,
            "realName": "name",
            "required": false,
            "type": "string",
          },
          Object {
            "model": undefined,
            "name": "title",
            "primaryKey": false,
            "realName": "title",
            "required": false,
            "type": "string",
          },
          Object {
            "model": undefined,
            "name": "num",
            "primaryKey": false,
            "realName": "num",
            "required": true,
            "type": "integer",
          },
          Object {
            "model": undefined,
            "name": "sum",
            "primaryKey": false,
            "realName": "sum",
            "required": false,
            "type": "float",
          },
          Object {
            "model": undefined,
            "name": "isActive",
            "primaryKey": false,
            "realName": "isActive",
            "required": true,
            "type": "boolean",
          },
          Object {
            "model": undefined,
            "name": "firstActive",
            "primaryKey": false,
            "realName": "firstActive",
            "required": true,
            "type": "date",
          },
          Object {
            "model": undefined,
            "name": "lastActive",
            "primaryKey": false,
            "realName": "lastActive",
            "required": false,
            "type": "date",
          },
          Object {
            "model": "model2",
            "name": "model2Field",
            "primaryKey": false,
            "realName": "model2Field",
            "required": false,
            "type": "model",
          },
          Object {
            "model": "model3",
            "name": "model3s",
            "primaryKey": false,
            "realName": "model3s",
            "required": false,
            "type": "collection",
          },
          Object {
            "model": undefined,
            "name": "id",
            "primaryKey": true,
            "realName": "id",
            "required": false,
            "type": "integer",
          },
          Object {
            "model": undefined,
            "name": "createdAt",
            "primaryKey": false,
            "realName": "createdAt",
            "required": false,
            "type": "date",
          },
          Object {
            "model": undefined,
            "name": "updatedAt",
            "primaryKey": false,
            "realName": "updatedAt",
            "required": false,
            "type": "date",
          },
        ],
        "id": "modelname1",
        "name": "ModelName1",
      },
      "id": "modelname1",
      "name": "ModelName1",
      "opts": Object {
        "interfaces": Array [],
      },
      "primaryKeyAttribute": Object {
        "model": undefined,
        "name": "_id",
        "primaryKey": true,
        "realName": "id",
        "required": false,
        "type": "integer",
      },
      "queryName": "modelName1",
      "resolveFn": undefined,
    },
  ],
}
`;
